### ISSUE TITLE
Please add support for gnuradio-3.9

### ISSUE THREAD

**Issue by yarda** [dl1ksv/gr-fcdproplus#23]
Gnuradio 3.9 is stable. It drops swig and uses pybind, so OOTs needs some changes. There is porting guide https://wiki.gnuradio.org/index.php/GNU_Radio_3.9_OOT_Module_Porting_Guide

We are trying to rebase gnuradio to 3.9 in Fedora Rawhide.

**Comment by dl1ksv** [comment]
There is a gr3.9 version of [fcdpp and fcd](https://github.com/dl1ksv/gr-funcube)
In <=gr3.7 fcd was part gnuradio. As it was removed first I put in my gr-fcdproplus repository. But as this name is msleading, I setup a fresh repository with upcoming 3.9. Can you use this repository ?

**Comment by yarda** [comment]
Great, thanks for info, so I will drop gr-fcdproplus and introduce new package gr-funcube to Fedora Rawhide (f34).

**Comment by yarda** [comment]
Is the API the same? I would like to add support to the osmosdr lib [1] and I wonder whether it's just matter of rename of includes/macros or whether it needs more effort.

[1] https://github.com/osmocom/gr-osmosdr/tree/master/lib/fcd

**Comment by dl1ksv** [comment]
As far I can see, this repo still assumes that gr-fcd is still a part of gnuradio.
I made a fork of [gr-osmosdr](https://github.com/dl1ksv/gr-osmosdr) gr-osmosdr last year, that is able to use gr-funcube .
Maybe I should update my fork.

**Comment by yarda** [comment]
It seems the repo is updated, it already has gr-3.9 support, I opened:
http://osmocom.org/issues/5031
I was also ready to work on the PR.

**Comment by dl1ksv** [comment]
Thanks for the  info. We'll see what happens.

**Comment by yarda** [comment]
If you have some patch or fork adding support to osmosdr, please do the PR or let me know, I would like to use it in Fedora downstream.

**Comment by dl1ksv** [comment]
Yes my fork supports gr-funcube and a plutosdr ( by gr-iio)  and I tested it together with gqrx, but I'm just working on some findPackage modules, to make run cmake smother.
I hope, to finis it in he next days.

**Comment by dl1ksv** [comment]
@yarda Meanwhile my fork of gr-osmosdr works. I could test it with the fcdpp, an rtl-sdr and and an adalm-plut

**Comment by yarda** [comment]
@dl1ksv thanks for info, I added your gr-osmosdr changes to Fedora downstream and linked it with gr-osmosdr upstream ticket.

Currently, I am dealing with some obscure gr-funcube crashes on s390x and ppc64le. It can be gnuradio related.

**Comment by yarda** [comment]
OK, I am maintaining it in Fedora, closing.
